src/components/DatePicker/Month/Month.tsx

```diff
@@ -41,12 +34,9 @@ export type Props = {
 
 const Month: React.FC<Props> = ({ year, month, onDaySelect, selectedDays, disabledDates }) => {
   const weeksWithDays = React.useMemo<WeekRow[]>(() => {
-    const monthDate = currentDay
-      .month(month)
-      .year(year)
-      .date(1);
+    const monthDate = currentDay.month(month).year(year).date(1);
     const daysOfMonth = monthDate.daysInMonth();
-    const startDay = monthDate.day();
+    const startDay = monthDate.day() ? monthDate.day() : 7;
```

Comment:
Repeating the comment made on the other PR so it doesn't get lost

Reply:
tbh, maybe it would be a better idea to update the reduce function to work with dayjs's indexing? It looks like it's deliberately changing the indexing from 0-6 to 1-7.

Would that be worth it or would it require more changes down the line?

Reply:
after a not-really-small investigation, refactoring the reduce logic would require many changes that would most probably affect the majority of the DatePicker's subcomponents. after a discussion with @panvourtsis we decided to use this fix for now